File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        
        int left =0, right;
        int numZero=0;
        int max=0;
        
        
        for(right=0; right<nums.length; right++) {
            
            if(nums[right]==0) {
                numZero++;
                // k--;
            }
            
            if(numZero>k) {
                
                if(nums[left]==0) numZero--;
                left++;
                
            }
            
            if(numZero<=k) {
                
                max = Math.max(max, right-left+1);
            }
        }
        
        return max;
        
    }
}

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        
        left, zero_num, res = 0,0,0
        
        for i, num in enumerate(nums):
            
            if num==0:
                zero_num+=1
                
            if zero_num>k:
                
                if nums[left]==0: 
                    zero_num-=1
                
                left+=1
            
            if zero_num<=k:
                res = max(res, i-left+1)
            
        return res
                    
                    
        